docs: Labware position check revisions

jwwojak opened the pull request:
# Overview

This is a change to Chapter 7, Software & Operations, Labware Position Check section. It includes new capabilities released w/ v8.4. New features or concepts include:

- Protocol independent nature of offsets, not tied to protocols
- Default offset concepts
- Applied offset concepts
- Hardcoded offsets

Includes new images and sections to support these changes.

Reference RTC-630.

## Test Plan and Hands on Testing

Checking for proper use of Markdown, HTML, and the usual writing stuff.

In follow-on updates, we ought to get these H3s moved up. There are a lot of H4s in this chapter and version.

Sandbox URL: [labware offsets and position checking](http://sandbox.docs.s3-website.us-east-2.amazonaws.com/labware-position-check-revisions/flex-manual/software-operation/#labware-offsets-and-position-checking)

## Changelog

The single section "Labware Position Check" becomes "Labware offsets and position checking."

- Main subsections "Labware offsets" and "Labware Position Check"
- New images, reusing old jog control images.

## Review requests

Also, @anthonywilleee might want to give this a look. 

## Risk assessment

Low, docs only. Could still be out of date as this was for 8.4. Can catch up in subsequent update.

jwwojak committed:
LPC changes for manual

Revises section.

jwwojak committed:
Update software-operation.md

jwwojak committed:
A few fixes

- replace smart quotes with straight quotes.

- proper name capitalize, "Labware Position Check"

- bold "Close" a UI control in the jog section.

- have vs had moment, but the original "have" is right here: "Flex requires that each piece of labware used in your protocol have a default or applied offset..."

jwwojak committed:
Revising touchscreen image

Image looked bad on large desk monitor, better on laptop.

Trying a different version with dot-end arrows docs typically use and exported at higher dpi. Check it on laptop monitor first.

jwwojak committed:
More labeled image testing

New image after trying several variations.

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,59 @@ When working with CSV files, keep in mind that:
-Labware Position Check guides you through these steps:
+| Feature | Description |
+|----|----|
+| Protocol independence | Offsets are labware-based and no longer bound to a 20-protocol limit. Each piece of labware requires one default offset and can also be associated with a significantly larger number of applied offsets. |
```
Technicality here: it was a _20-run_ limit, which was even worse than 20 unique protocols.

Update to say "20-run".

jwwojak replied:
Fixed in recent commits.

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,59 @@ When working with CSV files, keep in mind that:
+| Feature | Description |
+|----|----|
+| Protocol independence | Offsets are labware-based and no longer bound to a 20-protocol limit. Each piece of labware requires one default offset and can also be associated with a significantly larger number of applied offsets. |
+| Default offsets | Default offsets are automatically applied to particular piece of labware and to all other identical labware, regardless of deck slot or protocol. This 'measure once, set everywhere' feature means you don’t have to check offsets for duplicate labware, which helps reduce protocol setup time and effort. |
```
Recommend double quotation marks or italics for `measure once, set everywhere`

jwwojak replied:
Changed to double-quotes.

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,59 @@ When working with CSV files, keep in mind that:
+| Protocol independence | Offsets are labware-based and no longer bound to a 20-protocol limit. Each piece of labware requires one default offset and can also be associated with a significantly larger number of applied offsets. |
+| Default offsets | Default offsets are automatically applied to particular piece of labware and to all other identical labware, regardless of deck slot or protocol. This 'measure once, set everywhere' feature means you don’t have to check offsets for duplicate labware, which helps reduce protocol setup time and effort. |
+| Applied offsets | Applied offsets override defaults for a specific piece of labware in a specific deck slot. You can use an applied offset with different protocols, but the labware and deck slot must be the same as the original applied offset. |
+| Hardcoded offsets | A hardcoded offset is an offset type typically created by advanced users via the Opentrons Python API. Because these offsets are defined in code (`set_offset`), you cannot change them from the touchscreen or Opentrons App. You’ll need to modify the API code for a protocol to change a hardcoded offset. See [Setting Labware Offsets](https://docs.opentrons.com/v2/new_advanced_r ...
```
A little more precise:
```suggestion
| Hardcoded offsets | A hardcoded offset is an offset type typically created by advanced users via the Opentrons Python API. Because these offsets are defined in code (`set_offset`), you cannot change them from the touchscreen or Opentrons App. You’ll need to modify the Python protocol file to change a hardcoded offset. See [Setting Labware Offsets](https://docs.opentrons.com/v2/new_advanced_running.html?highlight=offset#setting-labware-offsets). |
```

jwwojak replied:
Will resolve in subsequent push.

ecormany commented on docs/flex-manual/docs/images/lpc-features-labeled.svg:
Typo in bottom left annotation needs to be fixed: `thedefault` -> `the default`.

jwwojak replied:
Fixed.

ecormany commented on docs/flex-manual/docs/images/lpc-missing-offsets.png:
Let's update this in Figma to match other screenshots in the chapter, with the protocol name "Nucleic Acid Purification with Magnetic Beads". And copy as a PNG at 2x, so the width should be exactly 1200px.

jwwojak replied:
Updated protocol title. Tried to get 1200px export. Removed "lpc" from image file name.

ecormany commented on docs/flex-manual/docs/images/lpc-features-labeled.svg:
I would recommend not using `lpc` in file names, since we don't use that abbreviation in the documentation itself.

jwwojak replied:
Removed "lpc" from image file name.

ecormany commented on docs/flex-manual/docs/images/lpc-labels.svg:
Oh, is this the newer, better version of this file? The other one is currently what's embedded in the page. Keep only one.

jwwojak replied:
I can't recall. Will sort out the images.

jwwojak replied:
Removed duplicate images and "lpc" from image file names.

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,59 @@ When working with CSV files, keep in mind that:
+This illustration shows how the different types of offsets appear as you're configuring a protocol on the Flex touchscreen.
 
-3.  Place a specific type of labware in a specific deck slot.
+<figure class="screenshot" markdown>
```
Might look nicer to omit `class="screenshot"` and let this image float in the page? Try it and see.

jwwojak replied:
Trying without. Looks fine either way.

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,59 @@ When working with CSV files, keep in mind that:
+Labware Position Check lets you align a pipette relative to a well plate (or other labware), which helps ensure accurate and reproducible pipetting results.
 
-7.  Remove the calibration probe from the pipette.
+Flex requires that each piece of labware used in your protocol have a default or applied offset associated with it. As shown in the touchscreen example below, you cannot run a protocol (the blue run button is inactive) if it uses labware that is missing offset data.
```
Agreement error. Should be either:
- "that each…has"
- "each…to have"

jwwojak replied:
I disagree. I think "have" is subjunctive and acceptable here regardless if the subject is singular or plural.

Can also say, "You must ensure that each piece of labware has a default or applied offset."

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,59 @@ When working with CSV files, keep in mind that:
+</figure>
 
-4.  Align the probe to the labware using the on-screen jog controls. Then confirm the position.
+You create offsets by running Labware Position Check on attached instruments and deck-mounted labware.
```
This phrasing feels off to me. Even in the new flow, you don't run Labware Position Check "on" a pipette. It uses a pipette, but its actions and output is completely with respect to labware. 

And we typically use the word "mount" for something that has been semi-permanently affixed to the robot, e.g. with screws, rather than labware that are placed into deck slots. "on-deck labware" would match our terminology better.

jwwojak replied:
Making changes. Anthony is in today and we'll check it out.

jwwojak replied:
Removing as unnecessary.

ecormany reviewed (changes requested):
Requesting a few prose changes for clarity and cleanup of image files.

jwwojak committed:
Some review comment changes

- added " " and removed ' ' from default offsets text in table.

- changed text in hardcoded offsets section of table.

- making recommended text changes throughout

- will try to fix image as requested in review.

jwwojak committed:
Fixing images and review comments

Add new offset UI image, remove images using LPC in name.

Minor text changes.

jwwojak committed:
Image adjustments

I'm not going to fight with these. Someone else can do this.

anthonywilleee commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,57 @@ When working with CSV files, keep in mind that:
-Labware Position Check guides you through these steps:
+| Feature | Description |
+|----|----|
+| Protocol independence | Offsets are labware-based and no longer bound to a 20-run limit. Each piece of labware requires one default offset and can also be associated with a significantly larger number of applied offsets. |
```
Should we get rid of the callout that they're no longer bound to the 20-protocol limit? Its describing an outdated technical constraint. It was a big change, but especially for new customers its kind of irrelevant.

I think the second sentence can stand to drive home the point of protocol independence a little more. suggestion:

> Offsets are labware-based and saved on your robot. Each piece of labware requires one default offset.  The default offset can be adjusted to create a significantly larger number of applied offsets. Since labware offset data is stored on your robot, it can be re-used in any protocol.

jwwojak replied:
Will revise in next push.

anthonywilleee commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,57 @@ When working with CSV files, keep in mind that:
+| Feature | Description |
+|----|----|
+| Protocol independence | Offsets are labware-based and no longer bound to a 20-run limit. Each piece of labware requires one default offset and can also be associated with a significantly larger number of applied offsets. |
+| Default offsets | Default offsets are automatically applied to particular piece of labware and to all other identical labware, regardless of deck slot or protocol. This "measure once, set everywhere" feature means you don’t have to check offsets for duplicate labware, which helps reduce protocol setup time and effort. |
```
At least for now, I think its important to call out that default offsets are manually created (and then can be automatically applied). This might change early next year if we automatically default to 0,0,0, but for now we should make it clear thats something someone has to do.  Rough suggestion:

> Default offsets are manually created using labware position check and then automatically applied to each instance of the labware, regardless of deck slot or protocol.  This 'measure once, set everywhere' feature means you don’t have to check offsets for duplicate labware, which helps reduce protocol setup time and effort.

anthonywilleee reviewed (commented):
Looks great! Left two comments

jwwojak committed:
A few changes

Reviewed text with Anthony locally and he left a review on Github.

jwwojak committed:
Review changes

Changing descriptions for the intro table, rows: protocol independence and default offsets.

Remove transition sentence after image, just start labware position check section.

anthonywilleee commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -216,42 +216,55 @@ When working with CSV files, keep in mind that:
+#### Labware Position Check
 
-5.  Repeat steps 3 and 4 for each labware--slot combination used in your protocol.
+Labware Position Check lets you align a gantry mounted instrument relative to a piece of labware (e.g. a well plate), which helps ensure accurate and reproducible pipetting results.
```
minor, but should we just say pipette instead of gantry mounted instrument? thats what youre actually using to run lpc

jwwojak replied:
Change to "pipette"

anthonywilleee reviewed (approved):
looks great! approved with one minor comment, up to you if you want to make that change

jwwojak committed:
LPC change from review

Change "gantry mounted instrument" to "pipette."

ecormany reviewed (approved):
Thanks @anthonywilleee and @jwwojak for the final back-and-forth to polish this section up! Content and implementation in mkdocs both look good.

ecormany committed:
Merge branch 'edge' into labware-position-check-revisions
